### PR TITLE
GH-1238: Fix permissions not throwing exception

### PR DESCRIPTION
**Pull request by jamesmontemagno**
We used to do this before using the new api, so need ot do checks.
Also put in some null checks on locaiton manager for good measuere

### Bugs Fixed ###

- Related to issue #1238

Provide links to issues here. Ensure that a GitHub issue was created for your feature or bug fix before sending PR.

### Behavioral Changes ###

Brings it back to expected behavior.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by Redth** [comment]
@jamesmontemagno maybe we should add an internal method that handles the throwing like we used to have instead?

**Comment by jamesmontemagno** [comment]
yes, let me do that :)

File: Xamarin.Essentials/Permissions/Permissions.ios.tvos.watchos.cs [hunk 1]
```diff
@@ -146,30 +150,44 @@ void LocationAuthCallback(object sender, CLAuthorizationChangedEventArgs e)
                     if (e.Status == CLAuthorizationStatus.NotDetermined)
                         return;
 
-                    if (previousState == CLAuthorizationStatus.AuthorizedWhenInUse && !whenInUse)
+                    try
```
Comment (mattleibow) [inline Xamarin.Essentials/Permissions/Permissions.ios.tvos.watchos.cs:153]: This double exception swallow is a bit scary... What is this for? Can we pass it to the tcs?

Comment (jamesmontemagno) [reply Xamarin.Essentials/Permissions/Permissions.ios.tvos.watchos.cs:153]: Yeah, I can do that especially if the tcs isn't null :)

Basically ... if you look at #1238 who knows that is happening so wanted to just catch it all!

File: Xamarin.Essentials/Permissions/Permissions.shared.cs [hunk 2]
```diff
@@ -16,6 +16,15 @@ internal static void EnsureDeclared<TPermission>()
             where TPermission : BasePermission, new() =>
                 new TPermission().EnsureDeclared();
 
+        internal static async Task RequestAndVerifyAsync<TPermission>()
```
Comment (mattleibow) [inline Xamarin.Essentials/Permissions/Permissions.shared.cs:19]: Can we change this to be `EnsureGrantedAsync` or something else Ensure-y to follow the pattern of check/request/ensure? I am never a fan of "and" in method names.

**Pull request merged**